### PR TITLE
FIX: sphinx >= 2.0 problems and more

### PR DESCRIPTION
**Pull request by jenisys**
FIXES:

* Sphinx >= 2.0 problems (tested with Sphinx v3.3.0)
* Tests while using newer pytest versions
* Add DIAG support, enable via environment variable: `SPHINXCONTRIB_ANSI_DIAG=yes`
* Works for python2.7 and python3 (python3.8 and python3.9 tested)

OTHERWISE:
* UPDATED: CHANGES.rst
* BUMP VERSION: v0.7.0

NOT FIXED:
* LICENSE file is missing in Github repository.

### CODE REVIEW

**Comment by ssbarnea** [comment]
Any chance to revive this work? Having ansi output in sphinx docs would be extremely helpful.

**Comment by jenisys** [comment]
Agreed.
Therefore, I provided this pull-request (to be merged) to fix the existing problems.
I prepared a release branch in my repository that at least I can have reproducible builds/installations with the provided/fixed functionality.

SEE ALSO:
* https://github.com/jenisys/sphinxcontrib-ansi/releases/tag/JE_v0.7.0 <https://github.com/jenisys/sphinxcontrib-ansi/releases/tag/JE_v0.7.0>

**Comment by ssbarnea** [comment]
@jenisys My impression was that you had maintainer rights on the repo, but I made a request for taking care of the project and unblock it. See https://github.com/sphinx-contrib/github-administration/issues/13

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**
